sdk/test/trace/tracer_provider_test.cc

```diff
@@ -0,0 +1,21 @@
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+
+#include <gtest/gtest.h>
+
+using opentelemetry::sdk::trace::TracerProvider;
+
+TEST(TracerProvider, GetTracer)
+{
+  auto tf = TracerProvider();
+  auto t1 = tf.GetTracer("test");
+  auto t2 = tf.GetTracer("test");
+  auto t3 = tf.GetTracer("different");
+  ASSERT_NE(t1, nullptr);
+  ASSERT_NE(t2, nullptr);
+  ASSERT_NE(t3, nullptr);
+
+  // Should return a new instance each time
```

Comment:
I thought the Tracer object was meant to be shared.

Reply:
I thought deduping based on name and version would be desired, as well. This is apparently not a requirement, and is an optimization that some folks are doing: https://github.com/open-telemetry/opentelemetry-cpp/pull/24#discussion_r363914305